daft/execution/udf.py

```diff
@@ -42,55 +46,50 @@ def read_and_release(self, name: str, size: int) -> bytes:
-            output_bytes = evaluated.to_ipc_stream()
-
-            out_name, out_size = transport.write_and_close(output_bytes)
-            conn.send(("success", out_name, out_size))
-    except Exception as e:
-        try:
-            conn.send(("error", type(e).__name__, traceback.format_exc()))
-        except Exception:
-            # If the connection is broken, it's because the parent process has died.
-            # We can just exit here.
-            pass
-    finally:
-        conn.close()
-
-
-class ActorHandle:
-    def __init__(self, projection: list[PyExpr]) -> None:
-        self.handle_conn, actor_conn = mp.Pipe(duplex=True)
-        expr_projection = ExpressionsProjection([Expression._from_pyexpr(expr) for expr in projection])
-        self.actor_process = mp.Process(target=actor_event_loop, args=(expr_projection, actor_conn), daemon=True)
-        self.actor_process.start()
+class UdfHandle:
+    def __init__(self, project_expr: PyExpr, passthrough_exprs: list[PyExpr]) -> None:
+        # Construct UNIX socket path for basic communication
+        with tempfile.NamedTemporaryFile(delete=True) as tmp:
+            self.socket_path = tmp.name
+        # Generate a random secret for establishing socket communication
+        secret = secrets.token_bytes(32)
+        # Initialize the main process listener
+        self.listener = Listener(self.socket_path, authkey=secret)
+
+        # Start the worker process
+        self.process = subprocess.Popen(
+            [
+                sys.executable,
+                "-m",
+                "daft.execution.udf_worker",
+                self.socket_path,
+                secret.hex(),
+            ]
+        )
+
+        # Initialize communication
+        self.handle_conn = self.listener.accept()
         self.transport = SharedMemoryTransport()
 
+        # Serialize and send the expression projection
+        expr_projection = ExpressionsProjection(
+            [Expression._from_pyexpr(expr) for expr in passthrough_exprs] + [Expression._from_pyexpr(project_expr)]
+        )
+        expr_projection_bytes = cloudpickle.dumps(expr_projection)
+        self.handle_conn.send(("__ENTER__", expr_projection_bytes))
+
     def eval_input(self, input: PyMicroPartition) -> PyMicroPartition:
-        if not self.actor_process.is_alive():
-            raise RuntimeError("Actor process is not alive")
+        if self.process.poll() is not None:
+            raise RuntimeError("UDF process has terminated")
 
         serialized = input.write_to_ipc_stream()
         shm_name, shm_size = self.transport.write_and_close(serialized)
         self.handle_conn.send((shm_name, shm_size))
 
         response = self.handle_conn.recv()
         if response[0] == "error":
-            error_type, tb_str = response[1], response[2]
-            exception_class = getattr(__builtins__, error_type, RuntimeError)
-            raise exception_class(tb_str)
```

Comment:
Originally, this would only correctly propagate errors up if they were builtin errors, and even then, the message of the error would change. I changed it to propagate the correct error and message, but no longer with the right traceback info. I can modify it to also contain the traceback string, wdty? Its not possible to send the original traceback without the `tblib` python library, I can potentially add that too?

Reply:
So what do we do? Should I include the `tblib` library?